src/app/modules/truenas-connect/components/truenas-connect-status-modal/truenas-connect-status-modal.component.ts

```diff
@@ -64,21 +69,40 @@ export class TruenasConnectStatusModalComponent {
 
   constructor(
     @Inject(WINDOW) private window: Window,
-    private matDialog: MatDialog,
     protected tnc: TruenasConnectService,
   ) { }
 
-  protected openSettings(): void {
-    this.matDialog
-      .open(TruenasConnectModalComponent, {
-        width: '456px',
-      })
-      .afterClosed()
-      .pipe(untilDestroyed(this))
-      .subscribe();
+  ngOnInit(): void {
+    // Automatically re-enable the service if it's disabled
+    if (this.tnc.config()?.status === TruenasConnectStatus.Disabled) {
+      this.tnc.enableService()
```

Comment:
This seems like it would be a blocking action. While this action is being performed, we should be showing a loading state to show the user that they can't use the UI while the service state changes. This just happens in the background. Also, allowing the user to perform other actions might cause this to error out without explanation or might make their new action fail without explanation if we are not deliberate with what's happening. Which might cause confusion. So we should have error handling here as well.